program/tests/test_create_mint.rs

```diff
@@ -1,256 +1,103 @@
-    let accounts = &[
-        (wrapped_mint_address, mint_account_with_data.clone()), // mint already exists
-        (wrapped_backpointer_address, backpointer_account_with_data), // backpointer already exists
-        (unwrapped_mint_address, Account::default()),
-        (solana_system_interface::program::id(), Account::default()),
-        mollusk_svm_programs_token::token2022::keyed_account(),
-    ];
-    mollusk.process_and_validate_instruction(&instruction, accounts, &[Check::success()]);
+    MintBuilder::default()
+        .idempotent()
+        .wrapped_mint_account(mint_account_with_data)
+        .backpointer_account(backpointer_account_with_data)
+        .execute();
 }
 
 #[test]
 fn test_idempotency_true_with_existing_invalid_accounts() {
-    let program_id = spl_token_wrap::id();
-    let mut mollusk = Mollusk::new(&program_id, "spl_token_wrap");
-    mollusk_svm_programs_token::token2022::add_program(&mut mollusk);
-
-    let unwrapped_mint_address = Pubkey::new_unique();
-    let wrapped_token_program_id = spl_token_2022::id();
-    let wrapped_mint_address =
-        get_wrapped_mint_address(&unwrapped_mint_address, &wrapped_token_program_id);
-    let wrapped_backpointer_address = get_wrapped_mint_backpointer_address(&wrapped_mint_address);
-
     // Incorrectly wrapped mint account owner
 
-    let instruction = create_mint(
-        &program_id,
-        &wrapped_mint_address,
-        &wrapped_backpointer_address,
-        &unwrapped_mint_address,
-        &wrapped_token_program_id,
-        true,
-    );
-
     let mint_account_with_data = Account {
         data: vec![1; 10],
         owner: Pubkey::new_unique(), // Wrong owner
         ..Account::default()
     };
-    let backpointer_account_with_data = Account {
-        owner: program_id,
-        ..Account::default()
-    };
 
-    let accounts = &[
-        (wrapped_mint_address, mint_account_with_data.clone()),
-        (wrapped_backpointer_address, backpointer_account_with_data),
-        (unwrapped_mint_address, Account::default()),
-        (solana_system_interface::program::id(), Account::default()),
-        mollusk_svm_programs_token::token2022::keyed_account(),
-    ];
-    mollusk.process_and_validate_instruction(
-        &instruction,
-        accounts,
-        &[Check::err(ProgramError::InvalidAccountOwner)],
-    );
+    MintBuilder::default()
+        .idempotent()
+        .wrapped_mint_account(mint_account_with_data)
+        .check(Check::err(ProgramError::InvalidAccountOwner))
+        .execute();
 
     // Incorrect owner on wrapped backpointer account
 
-    let instruction = create_mint(
-        &program_id,
-        &wrapped_mint_address,
-        &wrapped_backpointer_address,
-        &unwrapped_mint_address,
-        &wrapped_token_program_id,
-        true,
-    );
-
     let mint_account_with_data = Account {
         data: vec![1; 10],
-        owner: wrapped_token_program_id,
+        owner: spl_token_wrap::id(),
```

Comment:
The comment only mentions that the backpointer account has an incorrect owner, but this has a wrong owner on the mint too -- since the incorrect owner on the mint was tested earlier, should this be correct?

Reply:
Ah, a test bug! This is a good argument for custom errors to be used to disambiguate as `ProgramError::InvalidAccountOwner` is used in both cases. This passes because it just repeats case with the wrong owner on the mint. Will fix!